tests/unit/amo/middleware/test_prefixMiddleware.js

```diff
@@ -288,4 +289,15 @@ describe(__filename, () => {
     );
     sinon.assert.calledWith(fakeRes.set, 'Cache-Control', ['max-age=31536000']);
   });
+
+  it('should render a 404 for urls that are both a client app and locale exception', () => {
+    const fakeReq = {
+      originalUrl: '/robots.txt',
+      headers: {},
+    };
+    const statusSpy = sinon.spy(fakeRes, 'status');
+    prefixMiddleware(fakeReq, fakeRes, fakeNext, { _config: fakeConfig });
+    sinon.assert.notCalled(fakeRes.set);
```

Comment:
It might be better to assert that `fakeRes.redirect` is not called, because that is the main thing that happens of consequence. This works, but seems like it's based on more arbitrary behaviour.